src/sqlfluff/core/parser/grammar/base.py

```diff
@@ -45,6 +45,13 @@ def wrapped_method(self, parse_context: ParseContext):
     return wrapped_method
 
 
+class BracketInfo(NamedTuple):

```

Comment:
@barrywhart do you mind? my mypy locally complained and in general I'm not sure about class definitions nested in other classes

Reply:
It's a pretty common thing in C++. Probably less common in Python. I did it mainly for scoping, since it is not expected to be used outside the `BaseGrammar` class. No big deal, though.

I'm curious, what was the message from MyPy? Was it saying the code was bad or wrong (like a linting kind of thing) or more of a MyPy error (like it couldn't process the code)? I'm guessing the latter?

I googled for "mypy nested class" and found a couple of open issues:
* https://github.com/python/mypy/issues/6393
* https://github.com/python/mypy/issues/8397